messagediff.go

```diff
@@ -163,7 +163,9 @@ func (d *Diff) diff(aVal, bVal reflect.Value, path Path) bool {
 		if reflect.DeepEqual(aVal.Interface(), bVal.Interface()) {
 			equal = true
 		} else {
-			d.Modified[&path] = bVal.Interface()
+			localPath := make(Path, len(path))
+			copy(localPath, path)
+			d.Modified[&localPath] = bVal.Interface()
```

Comment:
Instead of just copying in that one specific location, would be good to move it to the top of the `diff` function. That way it fixes all cases of that.

``` go
localPath := make(Path, len(path))
copy(localPath, path)
path = localPath
```

Near https://github.com/d4l3k/messagediff/pull/8/files#diff-2aea0ca6be2f84668b7aefadf8f8ac26R44.